src/main/java/com/celements/course/service/ICourseServiceRole.java

```diff
@@ -32,4 +35,8 @@ public boolean validateParticipant(DocumentReference regDocRef, String emailAdr,
   @NotNull
   public CourseConfirmState getConfirmState(@NotNull DocumentReference regDocRef);
 
+  @NotNull
+  public List<EntityReference> getAnnouncementsForCourse(SpaceReference regSpaceRef,
+      DocumentReference partiClassRef);
```

Comment:
- NotNull/Nullable missing on parameters
- return `List<DocumentReference>`, the elements are always documents
- I don't think `partiClassRef` should be part of signature
- add `List<String> sortFields` to signature, makes it more reusable